### DISCUSSION TITLE
Zoom to particular position based on search result of editor text

### DISCUSSION THREAD

**Discussion by abhinay4512** [chrvadala/react-svg-pan-zoom#194] [Q&A] [answered]
Hi @chrvadala,
We are using react-svg-pan-zoom for pan and zoom functions. Currently we are trying to implement a search functionality on our svg editor. For which we have to zoom into the specific position on the editor based on the search result. We are getting x and y coordinates and also the transform matrix from the search result object and we are trying to use the setPointOnViewerCenter method for zooming in to the position. But we are having trouble finding the exact position using the x, y and matrix. It will be of great help if you could gives us some inputs on how to resolve this or point us to any documentation regarding this.
Thanks in Advance.

**Comment by chrvadala** [comment, marked as answer]
Hi guys, did you try the `fitSelection()` method. This functionality accepts a rect of the area that you want to show and library will arrange zoom and pan automatically. You can provide your image coords and the library will perform the hard calculation for you, so you don't have to worry about the math behind the matrix.

An example is available here https://chrvadala.github.io/react-svg-pan-zoom/?path=/story/react-svg-pan-zoom--call-methods
